### DISCUSSION TITLE
private-tmp with more argvs needed

### DISCUSSION THREAD

**Discussion by osevan** [netblue30/firejail#5104] [Q&A] [answered]
ApplicationA should got
private-tmp applicationsA.sock,applicationsB.sock

At the moment not possible

**Comment by rusty-snake** [comment, marked as answer]
```
whitelist /tmp/applicationsA.sock
whitelist /tmp/applicationsB.sock
private-tmp
```

**Reply by rusty-snake** [reply]
If you have an FR, open an issue.

**Reply by osevan** [reply]
Thank you very much

**Comment by osevan** [comment]
I see its working. 

But my usecases is:

ApplicationA and B need connection each other over sockets.

With private-tmp, hostside cant see sockets, because its running inside jail with private-tmp.

How I can connect to jails running application with unix sockets each other, and every other socket is banned.

I tried

blacklist /tmp/*.sock
whitelist /tmp/applicationA.sock
whitelist /tmp/applicationB.sock

But not working.

**Reply by rusty-snake** [reply]
You mean you run `firejail --whitelist=/tmp/foo.sock --private-tmp foo` (and `foo` creates `/tmp/foo.sock`) and then you run `firejail --whitelist=/tmp/foo.sock --private-tmp bar` and want to have /tmp/foo.sock inside bar's jail?

Solutions:
1. `whitelist /tmp/foo-sockets/foo.sock` and say `foo`,`bar` to use this path instead (maybe with `TMPDIR=/tmp/foo-sockets`)
2. If 1. is not possible, you have to drop `private-tmp` and `whitelist /tmp/…`.

**Reply by osevan** [reply]
Where I place TMPDIR=/tmp/foo-socket? 

In profile?

**Reply by osevan** [reply]
When i drop private-tmp its working well.

But i need private-tmp  to blacklist system sockets by default

**Reply by rusty-snake** [reply]
> But i need private-tmp to blacklist system sockets by default

`blacklist`

> Where I place TMPDIR=/tmp/foo-socket?

It's an envvar. How you set it does not matter.

This was an example, you need to check if your programs support it or if there are other ways to change the socket path.

**Reply by osevan** [reply]
With blacklist method, I need type every socket name by hand, it isn't effective.


But easy sharing sockets each other in /tmp/ will je nice,instead blacklist in profile 10 sockets everytime.

**Reply by rusty-snake** [reply]
```
noblacklist /tmp/foo
noblacklist /tmp/bar
blacklist /tmp/*
```
